File: app/src/main/java/com/huhx0015/androidbooster/injections/components/ServiceComponent.java
package com.huhx0015.androidbooster.injections.components;

import com.huhx0015.androidbooster.injections.modules.LocationModule;
import com.huhx0015.androidbooster.injections.scopes.ServiceScope;
import dagger.Component;

@ServiceScope
@Component(modules = {
        LocationModule.class
},
        dependencies = ApplicationComponent.class)
public interface ServiceComponent {}

File: app/src/main/java/com/huhx0015/androidbooster/injections/components/ServiceComponent.kt
package com.huhx0015.androidbooster.injections.components

import com.huhx0015.androidbooster.injections.modules.LocationModule
import com.huhx0015.androidbooster.injections.scopes.ServiceScope
import javax.inject.Singleton
import dagger.Component

@ServiceScope
@Singleton
@Component(
        modules = [LocationModule::class],
        dependencies = [ApplicationComponent::class])
interface ServiceComponent